src/app/WriteHandler.cpp

```diff
@@ -166,7 +167,23 @@ CHIP_ERROR WriteHandler::ProcessAttributeDataIBs(TLV::TLVReader & aAttributeData
             MatterPreAttributeWriteCallback(concretePath);
             TLV::TLVWriter backup;
             mWriteResponseBuilder.Checkpoint(backup);
-            err = WriteSingleClusterData(subjectDescriptor, clusterInfo, dataReader, this);
+            err = element.GetDataVersion(&requiredVersion);
+            if (CHIP_NO_ERROR == err)
+            {
+                if (!IsClusterDataVersionEqual(clusterInfo.mEndpointId, clusterInfo.mClusterId, requiredVersion))
```

Comment:
Per spec, the data version check happens last.  Here it's being done before the ACL checks, the "does this attribute even exist" check, etc.

Please put this check in the right place.

Reply:
agree, done, thanks